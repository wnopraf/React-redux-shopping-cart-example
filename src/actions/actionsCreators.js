import { ADD_PRODUCT, DELETE_PRODUCT, INCREMENT_SOTCK, DECREMENT_SOTCK, INCREMENT_AMOUNT, DECREMENT_AMOUNT } from '../constants'

export const AddProduct = (id) => {
  return {
    type: ADD_PRODUCT,
    id
  }
}

export const deleteProduct = (id) => {
  return {
    type: DELETE_PRODUCT,
    id
  }
}

const incrementStock = (id) => ({ type: INCREMENT_SOTCK, id })

const decrementStock = (id) => ({ type: DECREMENT_SOTCK, id })

const incrementAmount = (id) => ({ type: INCREMENT_AMOUNT, id })

const decrementAmount = (id) => ({ type: DECREMENT_AMOUNT, id })

export const incrementAction = (dispatch, getState) => (id) => {
  dispatch(decrementStock(id))
  const { cart: { amount } } = getState()
  selectIdItem(amount, id).amount > 0 && dispatch(incrementAmount(id))
}
export const decrementAction = (dispatch, getState) => (id) => {
  dispatch(incrementStock(id))
  dispatch(decrementAmount(id))
}
const selectIdItem = (stateSlice, id) => stateSlice.find(e => e.id === id)
