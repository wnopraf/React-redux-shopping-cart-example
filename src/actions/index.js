import { INCREMENT_SOTCK, DECREMENT_SOTCK, INCREMENT_AMOUNT, DECREMENT_AMOUNT } from '../constants'

const incrementStock = (id) => ({ type: INCREMENT_SOTCK, id })

const decrementStock = (id) => ({ type: DECREMENT_SOTCK, id })

const incrementAmount = (id) => ({ type: INCREMENT_AMOUNT, id })

const decrementAmount = (id) => ({ type: DECREMENT_AMOUNT, id })

export const incrementAction = (dispatch, getState) => (id) => {
  const { stock } = getState()
  if (selectIdItem(stock, id).stock > 0) {
    dispatch(decrementStock(id))
    dispatch(incrementAmount(id))
  }
}
export const decrementAction = (dispatch, getState) => (id) => {
  dispatch(incrementStock(id))
  dispatch(decrementAmount(id))
}
const selectIdItem = (stateSlice, id) => stateSlice.find(e => e.productId === id)
