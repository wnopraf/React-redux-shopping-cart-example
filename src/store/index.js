import { createStore } from 'redux'
import { stock, products } from '../data/products'
import cartReducer from '../reducers'

createStore(cartReducer, { products, stock })
