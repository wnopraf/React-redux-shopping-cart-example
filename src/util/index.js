export const selectIdItem = (stateSlice, id) => stateSlice.find(e => e.productId === id)
