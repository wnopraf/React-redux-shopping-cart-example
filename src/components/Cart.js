import React, { Component } from 'react'
import { incrementAction, decrementAction } from '../actions'
import { selectProductItem, selectIdItem } from '../util'
import { connect } from 'react-redux'

const CartItem = ({ products, cartItem, add, subtract, stock }) => {
  const productItem = selectProductItem(products, cartItem.productId)
  const stockItem = selectIdItem(stock, cartItem.productId)
  return <div className='cart-item' data-productId={cartItem.productId}>
    <div className='item-view'>
      <div className='item__features'>
        <h5>{productItem.name}</h5>
      </div>
    </div>
    <div className='item-value'>
      <div className='item__prize'>
        <span className='prize'>prize</span>
        <span className='value'>{productItem.prize} x {cartItem.amount} = {(productItem.prize * cartItem.amount).toFixed(2)}</span>
      </div>
      <div className='item-buttons'>
        <span className='units'>units</span>
        <span className='buttons'>
          <button data-add className={'add' + (stockItem.stock === 0 ? 'disabled' : '')} onClick={add}>+</button>
          <span className='amount'>{cartItem.amount}</span>
          <button data-subtract className='subtract' onClick={subtract}>-</button>
        </span>
      </div>
      {stockItem.stock === 0 && <div className='no-stock'>Out of existences</div>}
    </div>
  </div>
}

class Cart extends Component {
  constructor (props) {
    super(props)
    this.state = { isOpen: true }
    this.shipmnetCost = 3
  }
  componentDidMount () {
    window.addEventListener('resize', this.onresize)
  }
  onresize () {
    return () => {
      if (window.innerWidth > 768) {
        this.setState({ isopen: true })
      } else {
        this.setState({ isOpen: false })
      }
    }
  }
  increment (id) {
    return (e) => {
      const stockItem = selectIdItem(this.props.stock, id)
      if (stockItem.stock > 0) {
        this.props.dispatch(incrementAction(id))
      }
    }
  }
  decrement (id) {
    return (e) => this.props.dispatch(decrementAction(id))
  }
  calcTotal () {
    const total = this.props.cart.reduce((acum, item) => {
      let productItem = selectProductItem(this.props.products, item.productId)
      let totalCost = Number.parseFloat(productItem.prize) * item.amount + acum
      return totalCost
    }, 0)
    return total
  }
  render () {
    return <div className='cart-wraper'>
      <span className='cart-icon' onClick={e => this.setState({ isOpen: !this.state.isOpen })}>&#128722;<sapn className='items'>{this.props.cart.length}</sapn></span>
      {this.isOpen && <div className='cart'>

        {this.props.cart.map((e, i) => <CartItem add={this.increment(e.productId)} subtract={this.decrement(e.productId)} cartItem={e} key={i} products={this.props.products} stock={this.props.stock} />)}
        {!this.props.cart.length && <p className='empty-cart'>The cart is empty</p>}
        <div className='items-total'>
          <span className='label'>Total cart value</span>
          <span className='value'>{this.calcTotal().toFixed(2)} $</span>
        </div>
        <div className='shipment'>
          <span className='label'>shipment cost</span>
          <span className='value'>{this.shipmnetCost.toFixed(2)} $</span>
        </div>
        <div className='order-total'>
          <span className='label'>total cost order</span>
          <span className='value'>{(this.calcTotal() + this.shipmnetCost).toFixed(2)} $</span>
        </div>

        <span className='checkout-button'>checkout</span>

      </div>}
      <style jsx>{`
        .cart-icon {
          cursor: pointer
          font-size: 1.6rem;
        }
      
      `}</style>
    </div>
  }
}
const mapStateToProps = state => ({ cart: state.cart, stock: state.stock, products: state.products })
export default connect(mapStateToProps)(Cart)
