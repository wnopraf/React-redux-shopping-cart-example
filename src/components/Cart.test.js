import React from 'react'
import Cart from './Cart'
import store from '../store'
import { incrementAction, decrementAction } from '../actions'
import { Provider } from 'react-redux'
import createComponent from 'react-unit'
import renderer from 'react-test-renderer'
import { selectIdItem } from '../util'

describe('Cart component elements', () => {
  let cart
  let cartDom
  beforeAll(() => {
    cart = renderer.create(<Provider store={store}><Cart /></Provider>)
    cartDom = cart.root
  })

  test('Cart has 0 item', () => {
    expect.assertions(2)
    expect(cart.toJSON()).toMatchSnapshot()
    expect(cartDom.findAllByProps({ className: 'cart-item' }).length).toBe(0)
  })
  test('Cart has 1 item', () => {
    store.dispatch(incrementAction(4))
    expect.assertions(2)

    expect(cart.toJSON()).toMatchSnapshot()
    expect(cartDom.findAllByProps({ className: 'cart-item' }).length).toBe(1)
  })
  test('Cart has 2 item', () => {
    expect.assertions(2)
    store.dispatch(incrementAction(7))
    expect(cart.toJSON()).toMatchSnapshot()
    expect(cartDom.findAllByProps({ className: 'cart-item' }).length).toBe(2)
  })
  test('Cart lost 1 item', () => {
    expect.assertions(3)
    store.dispatch(decrementAction(4))
    expect(cart.toJSON()).toMatchSnapshot()
    expect(cartDom.findAllByProps({ className: 'cart-item' }).length).toBe(1)
    expect(cartDom.findAll(e => e.props['data-productId'] === 4).length).toBe(0)
  })
  describe('Solo item test', () => {
    let item_5
    beforeAll(() => {
      store.dispatch(incrementAction(5))
      item_5 = cartDom.find(e => e.props['data-productId'] === 5)
    })
    test('CartItem 5 has amount 2', () => {
      expect.assertions(2)

      item_5.find(e => e.props['data-add']).props.onClick()
      expect(cart.toJSON()).toMatchSnapshot()
      expect(item_5.find(e => e.props.className === 'amount').children[0]).toBe('2')
    })
    test('CartItem 5 has amount 3', () => {
      expect.assertions(2)
      item_5.find(e => e.props['data-add']).props.onClick()
      expect(cart.toJSON()).toMatchSnapshot()
      expect(item_5.find(e => e.props.className === 'amount').children[0]).toBe('3')
    })
    test('CartItem 5 has amount 2', () => {
      expect.assertions(2)
      item_5.find(e => e.props.className === 'subtract').props.onClick()
      expect(cart.toJSON()).toMatchSnapshot()
      expect(item_5.find(e => e.props.className === 'amount').children[0]).toBe('2')
    })
    test('CartItem 5 has no stock', () => {
      let stockItem = selectIdItem(store.getState().stock, 5)
      while (stockItem.stock > 0) {
        store.dispatch(incrementAction(5))
        stockItem = selectIdItem(store.getState().stock, 5)
      }
      item_5.find(e => e.props['data-add']).props.onClick()
      expect(cart.toJSON()).toMatchSnapshot()
      expect(item_5.findAll(e => e.props.className === 'no-stock').length).toBe(1)
    })
  })
})
