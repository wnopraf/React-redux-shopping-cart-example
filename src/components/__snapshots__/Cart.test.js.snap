// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart component elements Cart has 0 item 1`] = `
<div
  className="cart"
>
  <p
    className="empty-cart"
  >
    The cart is empty
  </p>
  <div
    className="items-total"
  >
    <span
      className="label"
    >
      Total cart value
    </span>
    <span
      className="value"
    >
      0.00
       $
    </span>
  </div>
  <div
    className="shipment"
  >
    <span
      className="label"
    >
      shipment cost
    </span>
    <span
      className="value"
    >
      3.00
       $
    </span>
  </div>
  <div
    className="order-total"
  >
    <span
      className="label"
    >
      total cost order
    </span>
    <span
      className="value"
    >
      3.00
       $
    </span>
  </div>
  <span
    className="checkout-button"
  >
    checkout
  </span>
</div>
`;

exports[`Cart component elements Cart has 1 item 1`] = `
<div
  className="cart"
>
  <div
    className="cart-item"
    data-productId={4}
  >
    <div
      className="item-view"
    >
      <div
        className="item__features"
      >
        <h5>
          product-D
        </h5>
      </div>
    </div>
    <div
      className="item-value"
    >
      <div
        className="item__prize"
      >
        <span
          className="prize"
        >
          prize
        </span>
        <span
          className="value"
        >
          135.66
           x 
          1
           = 
          135.66
        </span>
      </div>
      <div
        className="item-buttons"
      >
        <span
          className="units"
        >
          units
        </span>
        <span
          className="buttons"
        >
          <button
            className="add"
            data-add={true}
            onClick={[Function]}
          >
            +
          </button>
          <span
            className="amount"
          >
            1
          </span>
          <button
            className="subtract"
            data-subtract={true}
            onClick={[Function]}
          >
            -
          </button>
        </span>
      </div>
    </div>
  </div>
  <div
    className="items-total"
  >
    <span
      className="label"
    >
      Total cart value
    </span>
    <span
      className="value"
    >
      135.66
       $
    </span>
  </div>
  <div
    className="shipment"
  >
    <span
      className="label"
    >
      shipment cost
    </span>
    <span
      className="value"
    >
      3.00
       $
    </span>
  </div>
  <div
    className="order-total"
  >
    <span
      className="label"
    >
      total cost order
    </span>
    <span
      className="value"
    >
      138.66
       $
    </span>
  </div>
  <span
    className="checkout-button"
  >
    checkout
  </span>
</div>
`;

exports[`Cart component elements Cart has 2 item 1`] = `
<div
  className="cart"
>
  <div
    className="cart-item"
    data-productId={4}
  >
    <div
      className="item-view"
    >
      <div
        className="item__features"
      >
        <h5>
          product-D
        </h5>
      </div>
    </div>
    <div
      className="item-value"
    >
      <div
        className="item__prize"
      >
        <span
          className="prize"
        >
          prize
        </span>
        <span
          className="value"
        >
          135.66
           x 
          1
           = 
          135.66
        </span>
      </div>
      <div
        className="item-buttons"
      >
        <span
          className="units"
        >
          units
        </span>
        <span
          className="buttons"
        >
          <button
            className="add"
            data-add={true}
            onClick={[Function]}
          >
            +
          </button>
          <span
            className="amount"
          >
            1
          </span>
          <button
            className="subtract"
            data-subtract={true}
            onClick={[Function]}
          >
            -
          </button>
        </span>
      </div>
    </div>
  </div>
  <div
    className="cart-item"
    data-productId={7}
  >
    <div
      className="item-view"
    >
      <div
        className="item__features"
      >
        <h5>
          product-G
        </h5>
      </div>
    </div>
    <div
      className="item-value"
    >
      <div
        className="item__prize"
      >
        <span
          className="prize"
        >
          prize
        </span>
        <span
          className="value"
        >
          112.90
           x 
          1
           = 
          112.90
        </span>
      </div>
      <div
        className="item-buttons"
      >
        <span
          className="units"
        >
          units
        </span>
        <span
          className="buttons"
        >
          <button
            className="add"
            data-add={true}
            onClick={[Function]}
          >
            +
          </button>
          <span
            className="amount"
          >
            1
          </span>
          <button
            className="subtract"
            data-subtract={true}
            onClick={[Function]}
          >
            -
          </button>
        </span>
      </div>
    </div>
  </div>
  <div
    className="items-total"
  >
    <span
      className="label"
    >
      Total cart value
    </span>
    <span
      className="value"
    >
      248.56
       $
    </span>
  </div>
  <div
    className="shipment"
  >
    <span
      className="label"
    >
      shipment cost
    </span>
    <span
      className="value"
    >
      3.00
       $
    </span>
  </div>
  <div
    className="order-total"
  >
    <span
      className="label"
    >
      total cost order
    </span>
    <span
      className="value"
    >
      251.56
       $
    </span>
  </div>
  <span
    className="checkout-button"
  >
    checkout
  </span>
</div>
`;

exports[`Cart component elements Cart lost 1 item 1`] = `
<div
  className="cart"
>
  <div
    className="cart-item"
    data-productId={7}
  >
    <div
      className="item-view"
    >
      <div
        className="item__features"
      >
        <h5>
          product-G
        </h5>
      </div>
    </div>
    <div
      className="item-value"
    >
      <div
        className="item__prize"
      >
        <span
          className="prize"
        >
          prize
        </span>
        <span
          className="value"
        >
          112.90
           x 
          1
           = 
          112.90
        </span>
      </div>
      <div
        className="item-buttons"
      >
        <span
          className="units"
        >
          units
        </span>
        <span
          className="buttons"
        >
          <button
            className="add"
            data-add={true}
            onClick={[Function]}
          >
            +
          </button>
          <span
            className="amount"
          >
            1
          </span>
          <button
            className="subtract"
            data-subtract={true}
            onClick={[Function]}
          >
            -
          </button>
        </span>
      </div>
    </div>
  </div>
  <div
    className="items-total"
  >
    <span
      className="label"
    >
      Total cart value
    </span>
    <span
      className="value"
    >
      112.90
       $
    </span>
  </div>
  <div
    className="shipment"
  >
    <span
      className="label"
    >
      shipment cost
    </span>
    <span
      className="value"
    >
      3.00
       $
    </span>
  </div>
  <div
    className="order-total"
  >
    <span
      className="label"
    >
      total cost order
    </span>
    <span
      className="value"
    >
      115.90
       $
    </span>
  </div>
  <span
    className="checkout-button"
  >
    checkout
  </span>
</div>
`;

exports[`Cart component elements Solo item test CartItem 5 has amount 2 1`] = `
<div
  className="cart"
>
  <div
    className="cart-item"
    data-productId={7}
  >
    <div
      className="item-view"
    >
      <div
        className="item__features"
      >
        <h5>
          product-G
        </h5>
      </div>
    </div>
    <div
      className="item-value"
    >
      <div
        className="item__prize"
      >
        <span
          className="prize"
        >
          prize
        </span>
        <span
          className="value"
        >
          112.90
           x 
          1
           = 
          112.90
        </span>
      </div>
      <div
        className="item-buttons"
      >
        <span
          className="units"
        >
          units
        </span>
        <span
          className="buttons"
        >
          <button
            className="add"
            data-add={true}
            onClick={[Function]}
          >
            +
          </button>
          <span
            className="amount"
          >
            1
          </span>
          <button
            className="subtract"
            data-subtract={true}
            onClick={[Function]}
          >
            -
          </button>
        </span>
      </div>
    </div>
  </div>
  <div
    className="cart-item"
    data-productId={5}
  >
    <div
      className="item-view"
    >
      <div
        className="item__features"
      >
        <h5>
          product-E
        </h5>
      </div>
    </div>
    <div
      className="item-value"
    >
      <div
        className="item__prize"
      >
        <span
          className="prize"
        >
          prize
        </span>
        <span
          className="value"
        >
          136.68
           x 
          2
           = 
          273.36
        </span>
      </div>
      <div
        className="item-buttons"
      >
        <span
          className="units"
        >
          units
        </span>
        <span
          className="buttons"
        >
          <button
            className="add"
            data-add={true}
            onClick={[Function]}
          >
            +
          </button>
          <span
            className="amount"
          >
            2
          </span>
          <button
            className="subtract"
            data-subtract={true}
            onClick={[Function]}
          >
            -
          </button>
        </span>
      </div>
    </div>
  </div>
  <div
    className="items-total"
  >
    <span
      className="label"
    >
      Total cart value
    </span>
    <span
      className="value"
    >
      386.26
       $
    </span>
  </div>
  <div
    className="shipment"
  >
    <span
      className="label"
    >
      shipment cost
    </span>
    <span
      className="value"
    >
      3.00
       $
    </span>
  </div>
  <div
    className="order-total"
  >
    <span
      className="label"
    >
      total cost order
    </span>
    <span
      className="value"
    >
      389.26
       $
    </span>
  </div>
  <span
    className="checkout-button"
  >
    checkout
  </span>
</div>
`;

exports[`Cart component elements Solo item test CartItem 5 has amount 2 2`] = `
<div
  className="cart"
>
  <div
    className="cart-item"
    data-productId={7}
  >
    <div
      className="item-view"
    >
      <div
        className="item__features"
      >
        <h5>
          product-G
        </h5>
      </div>
    </div>
    <div
      className="item-value"
    >
      <div
        className="item__prize"
      >
        <span
          className="prize"
        >
          prize
        </span>
        <span
          className="value"
        >
          112.90
           x 
          1
           = 
          112.90
        </span>
      </div>
      <div
        className="item-buttons"
      >
        <span
          className="units"
        >
          units
        </span>
        <span
          className="buttons"
        >
          <button
            className="add"
            data-add={true}
            onClick={[Function]}
          >
            +
          </button>
          <span
            className="amount"
          >
            1
          </span>
          <button
            className="subtract"
            data-subtract={true}
            onClick={[Function]}
          >
            -
          </button>
        </span>
      </div>
    </div>
  </div>
  <div
    className="cart-item"
    data-productId={5}
  >
    <div
      className="item-view"
    >
      <div
        className="item__features"
      >
        <h5>
          product-E
        </h5>
      </div>
    </div>
    <div
      className="item-value"
    >
      <div
        className="item__prize"
      >
        <span
          className="prize"
        >
          prize
        </span>
        <span
          className="value"
        >
          136.68
           x 
          2
           = 
          273.36
        </span>
      </div>
      <div
        className="item-buttons"
      >
        <span
          className="units"
        >
          units
        </span>
        <span
          className="buttons"
        >
          <button
            className="add"
            data-add={true}
            onClick={[Function]}
          >
            +
          </button>
          <span
            className="amount"
          >
            2
          </span>
          <button
            className="subtract"
            data-subtract={true}
            onClick={[Function]}
          >
            -
          </button>
        </span>
      </div>
    </div>
  </div>
  <div
    className="items-total"
  >
    <span
      className="label"
    >
      Total cart value
    </span>
    <span
      className="value"
    >
      386.26
       $
    </span>
  </div>
  <div
    className="shipment"
  >
    <span
      className="label"
    >
      shipment cost
    </span>
    <span
      className="value"
    >
      3.00
       $
    </span>
  </div>
  <div
    className="order-total"
  >
    <span
      className="label"
    >
      total cost order
    </span>
    <span
      className="value"
    >
      389.26
       $
    </span>
  </div>
  <span
    className="checkout-button"
  >
    checkout
  </span>
</div>
`;

exports[`Cart component elements Solo item test CartItem 5 has amount 3 1`] = `
<div
  className="cart"
>
  <div
    className="cart-item"
    data-productId={7}
  >
    <div
      className="item-view"
    >
      <div
        className="item__features"
      >
        <h5>
          product-G
        </h5>
      </div>
    </div>
    <div
      className="item-value"
    >
      <div
        className="item__prize"
      >
        <span
          className="prize"
        >
          prize
        </span>
        <span
          className="value"
        >
          112.90
           x 
          1
           = 
          112.90
        </span>
      </div>
      <div
        className="item-buttons"
      >
        <span
          className="units"
        >
          units
        </span>
        <span
          className="buttons"
        >
          <button
            className="add"
            data-add={true}
            onClick={[Function]}
          >
            +
          </button>
          <span
            className="amount"
          >
            1
          </span>
          <button
            className="subtract"
            data-subtract={true}
            onClick={[Function]}
          >
            -
          </button>
        </span>
      </div>
    </div>
  </div>
  <div
    className="cart-item"
    data-productId={5}
  >
    <div
      className="item-view"
    >
      <div
        className="item__features"
      >
        <h5>
          product-E
        </h5>
      </div>
    </div>
    <div
      className="item-value"
    >
      <div
        className="item__prize"
      >
        <span
          className="prize"
        >
          prize
        </span>
        <span
          className="value"
        >
          136.68
           x 
          3
           = 
          410.04
        </span>
      </div>
      <div
        className="item-buttons"
      >
        <span
          className="units"
        >
          units
        </span>
        <span
          className="buttons"
        >
          <button
            className="add"
            data-add={true}
            onClick={[Function]}
          >
            +
          </button>
          <span
            className="amount"
          >
            3
          </span>
          <button
            className="subtract"
            data-subtract={true}
            onClick={[Function]}
          >
            -
          </button>
        </span>
      </div>
    </div>
  </div>
  <div
    className="items-total"
  >
    <span
      className="label"
    >
      Total cart value
    </span>
    <span
      className="value"
    >
      522.94
       $
    </span>
  </div>
  <div
    className="shipment"
  >
    <span
      className="label"
    >
      shipment cost
    </span>
    <span
      className="value"
    >
      3.00
       $
    </span>
  </div>
  <div
    className="order-total"
  >
    <span
      className="label"
    >
      total cost order
    </span>
    <span
      className="value"
    >
      525.94
       $
    </span>
  </div>
  <span
    className="checkout-button"
  >
    checkout
  </span>
</div>
`;

exports[`Cart component elements Solo item test CartItem 5 has no stock 1`] = `
<div
  className="cart"
>
  <div
    className="cart-item"
    data-productId={7}
  >
    <div
      className="item-view"
    >
      <div
        className="item__features"
      >
        <h5>
          product-G
        </h5>
      </div>
    </div>
    <div
      className="item-value"
    >
      <div
        className="item__prize"
      >
        <span
          className="prize"
        >
          prize
        </span>
        <span
          className="value"
        >
          112.90
           x 
          1
           = 
          112.90
        </span>
      </div>
      <div
        className="item-buttons"
      >
        <span
          className="units"
        >
          units
        </span>
        <span
          className="buttons"
        >
          <button
            className="add"
            data-add={true}
            onClick={[Function]}
          >
            +
          </button>
          <span
            className="amount"
          >
            1
          </span>
          <button
            className="subtract"
            data-subtract={true}
            onClick={[Function]}
          >
            -
          </button>
        </span>
      </div>
    </div>
  </div>
  <div
    className="cart-item"
    data-productId={5}
  >
    <div
      className="item-view"
    >
      <div
        className="item__features"
      >
        <h5>
          product-E
        </h5>
      </div>
    </div>
    <div
      className="item-value"
    >
      <div
        className="item__prize"
      >
        <span
          className="prize"
        >
          prize
        </span>
        <span
          className="value"
        >
          136.68
           x 
          13
           = 
          1776.84
        </span>
      </div>
      <div
        className="item-buttons"
      >
        <span
          className="units"
        >
          units
        </span>
        <span
          className="buttons"
        >
          <button
            className="adddisabled"
            data-add={true}
            onClick={[Function]}
          >
            +
          </button>
          <span
            className="amount"
          >
            13
          </span>
          <button
            className="subtract"
            data-subtract={true}
            onClick={[Function]}
          >
            -
          </button>
        </span>
      </div>
      <div
        className="no-stock"
      >
        Out of existences
      </div>
    </div>
  </div>
  <div
    className="items-total"
  >
    <span
      className="label"
    >
      Total cart value
    </span>
    <span
      className="value"
    >
      1889.74
       $
    </span>
  </div>
  <div
    className="shipment"
  >
    <span
      className="label"
    >
      shipment cost
    </span>
    <span
      className="value"
    >
      3.00
       $
    </span>
  </div>
  <div
    className="order-total"
  >
    <span
      className="label"
    >
      total cost order
    </span>
    <span
      className="value"
    >
      1892.74
       $
    </span>
  </div>
  <span
    className="checkout-button"
  >
    checkout
  </span>
</div>
`;
